src/index.js

```diff
@@ -0,0 +1,194 @@
+      } else {
+        throw new Error('WrappedComponent lacks a handleClickOutside(event) function for processing outside click events.');
+      }
+
+      // TODO: try to get rid of this, could be done with function ref, might be problematic for SFC though, they do not expose refs
+      const componentNode = this.instanceRef;
+      if (componentNode === null) {
+        console.warn('Antipattern warning: there was no DOM node associated with the component that is being wrapped by outsideClick.');
+        console.warn([
+          'This is typically caused by having a component that starts life with a render function that',
+          'returns `null` (due to a state or props value), so that the component \'exist\' in the React',
+          'chain of components, but not in the DOM.\n\nInstead, you need to refactor your code so that the',
+          'decision of whether or not to show your component is handled by the parent, in their render()',
+          'function.\n\nIn code, rather than:\n\n  A{render(){return check? <.../> : null;}\n  B{render(){<A check=... />}\n\nmake sure that you',
+          'use:\n\n  A{render(){return <.../>}\n  B{render(){return <...>{ check ? <A/> : null }<...>}}\n\nThat is:',
+          'the parent is always responsible for deciding whether or not to render any of its children.',
+          'It is not the child\'s responsibility to decide whether a render instruction from above should',
+          'get ignored or not by returning `null`.\n\nWhen any component gets its render() function called,',
+          'that is the signal that it should be rendering its part of the UI. It may in turn decide not to',
+          'render all of *its* children, but it should never return `null` for itself. It is not responsible',
+          'for that decision.'
+        ].join(' '));
+      }
+
+      const fn = this.__outsideClickHandler = generateOutsideCheck(
+        componentNode,
+        instance,
+        clickOutsideHandler,
+        this.props.outsideClickIgnoreClass,
+        this.props.excludeScrollbar,
+        this.props.preventDefault,
+        this.props.stopPropagation
+      );
+
+      const pos = registeredComponents.length;
+      registeredComponents.push(this);
+      handlers[pos] = fn;
+
+      // If there is a truthy disableOnClickOutside property for this
+      // component, don't immediately start listening for outside events.
+      if (!this.props.disableOnClickOutside) {
+        this.enableOnClickOutside();
+      }
+    }
+
+    /**
+    * Track for disableOnClickOutside props changes and enable/disable click outside
+    */
+    componentWillReceiveProps(nextProps) {
+      if (this.props.disableOnClickOutside && !nextProps.disableOnClickOutside) {
+        this.enableOnClickOutside();
+      } else if (!this.props.disableOnClickOutside && nextProps.disableOnClickOutside) {
+        this.disableOnClickOutside();
+      }
+    }
+
+    /**
+     * Remove the document's event listeners
```

Comment:
Just for correctness, "Remove all document's event listeners for this component"